### PR TITLE
Enrichment with txpool tests

### PR DESCRIPTION
**Pull request by novosandara**
# Description

Please provide a detailed description of what was done in this PR

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes
# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

# Documentation update

# Additional comments

### CODE REVIEW

File: txpool/txpool_test.go [hunk 1]
```diff
@@ -321,6 +342,23 @@ func TestAddTxErrors(t *testing.T) {
...
+		pool := setupPool()
+
+		// fill the pool leaving only 1 slot
+		pool.gauge.increase(defaultMaxSlots - 1)
+
+		// create tx requiring 1 slot
+		tx := newTx(defaultAddr, 0, 1, types.LegacyTxType)
+		tx = signTx(tx)
+
+		//	enqueue tx
```
Comment (Stefan-Ethernal) [inline txpool/txpool_test.go]:
```suggestion
		// enqueue tx
```

File: consensus/polybft/proto/transport.pb.validate.go [hunk 2]
```diff
@@ -39,18 +39,18 @@ var (
 // in the proto definition for this message. If any rules are violated, the
 // first error encountered is returned, or nil if there are no violations.
 func (m *TransportMessage) Validate() error {
-	return m.validate(false)
+	return m.validate()
```
Comment (Stefan-Ethernal) [inline consensus/polybft/proto/transport.pb.validate.go]: This code is auto-generated, so you should probably revert these changes, in case they aren't made when you re-run the protobuf compiler.

File: txpool/txpool_test.go [hunk 3]
```diff
@@ -321,6 +342,23 @@ func TestAddTxErrors(t *testing.T) {
...
+		assert.NoError(t, pool.AddTx(tx))
+		<-pool.promoteReqCh
```
Comment (Stefan-Ethernal) [inline txpool/txpool_test.go:362]: Does it make sense to assert the length of the tx pool at this point?

File: txpool/txpool_test.go [hunk 4]
```diff
@@ -4042,63 +4229,139 @@ func getDefaultEnabledForks() *chain.Forks {
...
+			pool.SetSigner(signer)
+
+			for i := 0; i < len(txs); i++ {
+				err = pool.addTx(local, txs[i])
+				require.NoError(b, err)
+			}
+		}
+	})
+
+	b.Run("benchmark more accounts more add transaction and transaction replacement", func(b *testing.B) {
```
Comment (Stefan-Ethernal) [inline txpool/txpool_test.go]:
```suggestion
	b.Run("benchmark multiple accounts add multiple transactions with transaction replacement", func(b *testing.B) {
```

File: txpool/txpool_test.go [hunk 5]
```diff
@@ -4042,63 +4229,139 @@ func getDefaultEnabledForks() *chain.Forks {
...
+			pool.SetSigner(signer)
+
+			for i := 0; i < len(txs); i++ {
+				err = pool.addTx(local, txs[i])
+				require.NoError(b, err)
+			}
+		}
+	})
+
+	b.Run("benchmark more accounts add more transactions", func(b *testing.B) {
```
Comment (Stefan-Ethernal) [inline txpool/txpool_test.go]:
```suggestion
	b.Run("benchmark multiple accounts add multiple transactions", func(b *testing.B) {
```

File: txpool/txpool_test.go [hunk 6]
```diff
@@ -4042,63 +4229,139 @@ func getDefaultEnabledForks() *chain.Forks {
...
+			i := (uint64)(0)
+			for i < defaultMaxAccountEnqueued {
+				txs[n*defaultMaxAccountEnqueued+i], err = signer.SignTx(newTx(addr, i, uint64(1), types.LegacyTxType), key)
+				require.NoError(b, err)
+
+				i++
 			}
+
+			n++
 		}
```
Comment (Stefan-Ethernal) [inline txpool/txpool_test.go]: Suggestion to simplify the loops:
```suggestion
		for n := uint64(0); n < accountNumber; n++ {
			key, err := crypto.GenerateECDSAPrivateKey()
			require.NoError(b, err)

			addr := crypto.PubKeyToAddress(&key.PublicKey)

			for i := uint64(0); i < defaultMaxAccountEnqueued; i++ {
				txs[n*defaultMaxAccountEnqueued+i], err = signer.SignTx(newTx(addr, i, uint64(1), types.LegacyTxType), key)
				require.NoError(b, err)
			}
		}
```

File: txpool/txpool_test.go (hunk 3 above)
Comment (novosandara) [reply txpool/txpool_test.go:362]: It can be checked if there is a newly added transaction in the index map.

**Review by Stefan-Ethernal — APPROVED** [review]

**Review by goran-ethernal — APPROVED** [review]

File: txpool/txpool_test.go [hunk 7]
```diff
@@ -3627,6 +3730,86 @@ func TestGetTxs(t *testing.T) {
 	}
 }
 
+func TestFreeSlots(t *testing.T) {
+	pool, err := newTestPool()
+	require.NoError(t, err)
+
+	freeslot := pool.gauge.freeSlots()
+	require.NotEqualValues(t, uint64(0), freeslot)
```
Comment (oliverbundalo) [inline txpool/txpool_test.go]:
```suggestion
	require.Greater(t, freeslot, uint64(0))
```
Maybe to use Greater instead of NotEqualValue. It is more appropriate for this test since we are expecting some value greater than 0.

File: txpool/txpool_test.go [hunk 8]
```diff
@@ -3627,6 +3730,86 @@ func TestGetTxs(t *testing.T) {
...
+	pool.SetSigner(signer)
+
+	addr := crypto.PubKeyToAddress(&key.PublicKey)
+	tx, err := signer.SignTx(newTx(addr, 0, 1, types.LegacyTxType), key)
+	require.NoError(t, err)
+
+	assert.NoError(t, pool.addTx(local, tx))
+
+	read, max := pool.GetCapacity()
+	assert.NotEqual(t, uint64(0), read)
```
Comment (oliverbundalo) [inline txpool/txpool_test.go]:
```suggestion
	assert.Greater(t, read, uint64(0))
```
The same here

File: txpool/txpool_test.go [hunk 9]
```diff
@@ -3627,6 +3730,86 @@ func TestGetTxs(t *testing.T) {
...
+	assert.NotEqual(t, uint64(0), max)
```
Comment (oliverbundalo) [inline txpool/txpool_test.go]:
```suggestion
	assert.Greater(t, max, uint64(0))
```

**Review by oliverbundalo — APPROVED** [review]
